lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -294,6 +294,7 @@ def headers(post, options = {})
           'Authorization' => signature(post, timestamp)
         }
         headers['X-Idempotency-Key'] = options[:idempotency_key] if options[:idempotency_key]
+        headers['X-Dlocal-Payment-Source'] = options[:application_id] if options[:application_id]
```

Comment:
This is not how application_id gets passed in. 

application_id is a class variable so it would be passed in like

```suggestion
        headers['X-Dlocal-Payment-Source'] = application_id if application_id
```